mlflow/types/schema.py

```diff
@@ -406,3 +458,280 @@ def __eq__(self, other) -> bool:
 
     def __repr__(self) -> str:
         return repr(self.inputs)
+
+
+@experimental
+class ParamSpec:
+    """
+    Specification used to represent parameters for the model.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        type: Union[DataType, str],  # pylint: disable=redefined-builtin
```

Comment:
Could we use a different name here? `param_type`? Overriding a reserved func can be a bit dangerous / hard to follow.

Reply:
+1

Reply:
or `date_type`?

Reply:
or `dtype`?

Reply:
TensorSpec and ColSpec uses type, do we want to keep consistent?

Reply:
I think those are in need of a refactoring later on. ;)